tfhe/src/core_crypto/gpu/entities/glwe_ciphertext_list.rs

```diff
@@ -68,6 +68,49 @@ impl<T: UnsignedInteger> CudaGlweCiphertextList<T> {
         Self(cuda_glwe_list)
     }
 
+    pub fn from_container<C: Container<Element = T>>(
+        container: &C,
+        glwe_size: GlweSize,
+        polynomial_size: PolynomialSize,
+        ciphertext_modulus: CiphertextModulus<T>,
+        streams: &CudaStreams,
+    ) -> Self {
+        assert!(
+            container.container_len() % glwe_ciphertext_size(glwe_size, polynomial_size) == 0,
+            "The provided container length is not valid. \
+        It needs to be dividable by glwe_size * polynomial_size. \
+        Got container length: {}, glwe_size: {glwe_size:?}, polynomial_size: {polynomial_size:?}.",
+            container.container_len()
+        );
+
+        let glwe_dimension = glwe_size.to_glwe_dimension();
+        let glwe_ciphertext_count = GlweCiphertextCount(
+            container.container_len() / glwe_ciphertext_size(glwe_size, polynomial_size),
+        );
+
+        let mut d_vec = CudaVec::new(
+            glwe_ciphertext_size(glwe_dimension.to_glwe_size(), polynomial_size)
+                * glwe_ciphertext_count.0,
+            streams,
+            0,
+        );
+        // Copy to the GPU
+        unsafe {
+            d_vec.copy_from_cpu_async(container.as_ref(), streams, 0);
+        }
+        streams.synchronize();
+
+        let cuda_glwe_list = CudaGlweList {
+            d_vec,
+            glwe_ciphertext_count,
+            glwe_dimension,
+            polynomial_size,
+            ciphertext_modulus,
+        };
+
+        Self(cuda_glwe_list)
+    }
```

Comment:
I don't think this is needed, a user should first convert its container to glwe_ciphertext_list then call `from_glwe_ciphertext_list`, or If you prefrer to keep this function, make this function call `GlweCiphertextList::from_container` then call `Self::from_glwe_ciphertext_list` so that we avoid having too much duplicated logic

Reply:
Indeed. I wanted to cut a corner but to have a `from_container()` method in `CudaGlweCiphertextList` can be confusing. I will remove it.